libnavigation-base/api/current.txt

```diff
@@ -1625,9 +1625,11 @@ package com.mapbox.navigation.base.trip.model.roadobject.reststop {
 
   public final class RestStop extends com.mapbox.navigation.base.trip.model.roadobject.RoadObject {
     method public java.util.List<com.mapbox.navigation.base.trip.model.roadobject.reststop.Amenity>? getAmenities();
+    method public String? getGuideMapUri();
```

Comment:
Interesting this has not been failing 🤔

Reply:
Maybe some check is off, I often encounter such cases lately.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/6237

Referring to where it was added.

Reply:
https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/20114/workflows/e589fceb-79f3-4db6-92cd-473c76749c80/jobs/99816

The job passed 🤔

Reply:
this is passing on main line, looks like an issue with `checkApi`

cc: @Guardiola31337

Reply:
Wait, isn't checkApi to track potential breaking changes? These changes were not breaking, they are just additions.

Reply:
Correct, that's exactly what I was going to say 😅

Reply:
The question is that this addition should have happened in that other PR, not in yours.

Reply:
Well, yes, but in that PR current.txt was not updated, so I updated it in mine because I ran the command and the change appeared, so why not? Or do you prefer fixing these kinds of things in separate PRs? I made a separate commit for that.

Reply:
That's fine, the question is as @kmadsen pointed out why `checkApi` didn't capture that in @RingerJK's PR forcing him to do so.

Reply:
Because it's not a breaking change? If I add a method to `MapboxNavigation`, do not update current.txt and run `make core-check-api` it does not fail. Because I didn't break anything. And if I remove a method or change its signature, it fails. That's how it's supposed to be, right? Or am I wrong?

Reply:
Nah, Metalava should fail on any change, warning the developer of any potential breaking change. I'm investigating as it's not failing for me either. As @kmadsen mentioned it may be a Metalava issue.

Reply:
Oh, I've always seen it work the way I described so I assumed it's supposed to be like this...

Reply:
I think I found the root cause. It's a "regression" from https://github.com/mapbox/mapbox-navigation-android/pull/5883

We've been using an old Metalava version for a while and the tool has changed considerably since then - proof of that is all the hiccups we run into when updating it https://github.com/mapbox/mapbox-navigation-android/pull/5883#issuecomment-1145326580 😬

Now, the `AddedMethod`s aren't reported as errors unless adding `'--error', 'AddedMethod',` to https://github.com/mapbox/mapbox-navigation-android/blob/4708d3c8255e1d3a84e53348cbd92c029d45140f/gradle/track-public-apis.gradle#L61 explicitly but if you do so you start getting inherited methods errors (these are now skipped by default, see https://android.googlesource.com/platform/tools/metalava/+/refs/heads/master/src/main/java/com/android/tools/metalava/CompatibilityCheck.kt#783) so I'd say that this is a feature, not a bug because technically adding a method doesn't break SemVer and we don't want the tool to report it as an error. This has one caveat which is that we may forget to update the baseline files and run into situations in which some breaking SemVer changes go unnoticed and therefore we introduce breaking changes unexpectedly. For that, I suggest leaving this as is (i.e. `AddedMethod`s aren't reported anymore as they don't technically break SemVer) but updating the baseline files from CI automatically if the check steps is successful, so they are always up-to-date and conflicts won't be an option. @mapbox/navigation-android @SevaZhukov what do you think? Will cut a ticket to keep track of this.

In any case, as mentioned here https://github.com/mapbox/mapbox-navigation-android/pull/5883#discussion_r888452547 I'm still wondering if now it's a good time to sunset Metalava and rely on something else.

> Yeah, wondering if we should keep using a Google / Android "internal" tool i.e. docs are poor and not well maintained, changes in the repo are really hard to follow, as it's still in alpha they keep breaking SemVer which makes updating it very difficult sometimes...
>
> We may want to consider something else 🤔
>
> Also noting that the Maps team already integrates [Binary compatibility validator](https://github.com/Kotlin/binary-compatibility-validator) https://github.com/mapbox/mapbox-maps-android/pull/1120.

Reply:
> Will cut a ticket to keep track of this.

https://github.com/mapbox/mapbox-navigation-android/issues/6257